packaging_automation/citus_package.py

```diff
@@ -210,15 +221,22 @@ def sign_packages(sub_folder: str, signing_credentials: SigningCredentials,
         print("DEB signing finished successfully.")
 
 
-def get_postgres_versions(os_name: str, input_files_dir: str) -> Tuple[List[str], List[str]]:
-    if platform_postgres_version_source[os_name] == PostgresVersionDockerImageType.single:
-        release_versions = ["all"]
-        nightly_versions = ["all"]
-    else:
-        package_version = get_package_version_without_release_stage_from_pkgvars(input_files_dir)
-        release_versions = get_supported_postgres_release_versions(f"{input_files_dir}/{POSTGRES_MATRIX_FILE_NAME}",
-                                                                   package_version)
-        nightly_versions = get_supported_postgres_nightly_versions(f"{input_files_dir}/{POSTGRES_MATRIX_FILE_NAME}")
+def get_postgres_versions(platform: str, input_files_dir: str) -> Tuple[List[str], List[str]]:
```

Comment:
I have trouble reading and understanding this file. Can we have a simpler logic for set difference maybe?

Reply:
One idea is to use set difference to remove the excluded items from supported items. I roughly have the following idea:

```py
>>> d1 = {1:1, 2:2, 3:3, "foo":4}
>>> d2 = {1:1, 2:4, 3:3, "bar":5, 6:6}
>>> dict(set(d1.items()).difference(set(d2.items())))
{'foo': 4, 2: 2}
```

Reply:
```suggestion

postgres_versions = [12, 13, 14]
exclude_dict = {"ol/7": [14]}
platform = "ol/7"

if platform in exclude_dict:
    filtered_postgres_versions = set(postgres_versions).difference(set(exclude_dict[platform]))
    print(filtered_postgres_versions)
```

```suggestion
release_versions = [v for v in release_versions if v not in exclude_dict_release[platform_key_release]]
```